pkg/controller/clusterdeployment/clusterdeployment_deleted.go

```diff
@@ -37,16 +43,35 @@ func (r *ReconcileClusterDeployment) handleDelete(request reconcile.Request, ins
 		return reconcile.Result{}, err
 	}
 
+	println("1")
+
 	err = pdData.ParseClusterConfig(r.client, request.Namespace, request.Name)
 	if err != nil {
+		println("1 err: " + err.Error())
 		return reconcile.Result{}, err
 	}
 
+	println("2")
+
 	err = r.pdclient.DeleteService(pdData)
 	if err != nil {
 		r.reqLogger.Error(err, "Failed cleaning up pagerduty.")
 	}
 
+	// find the PD syncset and delete it
+	ssName := request.Name + config.SyncSetPostfix
+	println(ssName)
+	r.reqLogger.Info("Deleting PD SyncSet", "Namespace", request.Namespace, "Name", request.Name)
+	syncset := &hivev1.SyncSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: ssName}, syncset)
+	if err == nil {
+		err = r.client.Delete(context.TODO(), syncset)
+		if err != nil {
+			r.reqLogger.Error(err, "Error deleting SyncSet", "Namespace", request.Namespace, "Name", ssName)
+			return reconcile.Result{}, err
```

Comment:
Couldn't there be a race condition here where the CR is updated after a deletion timestamp?

If it can't delete it wouldn't it be better to let it be until the NS gets cleaned up?

Reply:
We don't fail if we can't delete the PD service.  I can see letting it fall through for that reason.